src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java

```diff
@@ -1815,16 +1815,29 @@ private String determineFilename(
         }
 
         final Date disbursementDate = extractTypeContext.getDisbursementDate();
-        final SimpleDateFormat sdf = new SimpleDateFormat("yyyyMMdd_HHmmss", Locale.US);
+        final String filename = buildUniqueFilename(directoryName, finalFormattedCheckFilePrefix, disbursementDate, null);
+
+        LOG.debug("determineFilename(...) - : filename={}", filename);
+        return filename;
+    }
+
+    private String buildUniqueFilename(String directoryName, String finalFormattedCheckFilePrefix, Date disbursementDate, Integer uniquenessSuffix) {
```

Comment:
Since this file is an overlay of base code, please add appropriate comments to point out this customization to it.